src/components/hooks/useForm/useForm.ts

```diff
@@ -26,16 +26,20 @@ export interface IUseFormValues {
 
 const useForm = ({ initialValues, validate, onSubmit }: IUseFormProps): IUseFormValues => {
   const [values, updateValues] = useState(initialValues);
-  const [errors, updateErrors] = useState(validate ? validate(initialValues) : {});
+  const [errors, updateErrors] = useState<Record<string, string | undefined>>({});
   const [touched, updateTouched] = useState<Record<string, boolean>>({});
 
-  const runValidation = (formValues: typeof initialValues) => {
+  const runValidation = async (formValues: typeof initialValues) => {
     if (validate) {
-      const errors = validate(formValues);
+      const errors = await validate(formValues);
       updateErrors(errors);
     }
   };
 
+  useEffect(() => {
+    runValidation(values);
+  }, []);
```

Comment:
Should we watch for `values` here? ( _to make sure validation is re-run when they update_ ) 💅🏻

Reply:
The validation runs `onChange` of the input so no need to. This is here just to run the validation on mount in case you the info is prefilled. Since the validation can by async now, I can't initialize the errors state with the result of `validate` anymore